16.0.1.1 flight phase views constraint

annzzu opened the pull request:
[IMP] flight_event: add phase duration views and constraints

- Add tree, form, and search views and menu item for flight.phase.duration model
- Implement constraint to ensure correct event time sequence per phase/flight
- Update README

16.0.1.1.0 version

annzzu committed:
[FIX] flight_event phase duration recalculation
- calculate flight block and flight time using ORM instead of SQL
- seperate model files and rename

16.0.1.0.

annzzu committed:
[IMP] flight_event: add phase duration views and constraints

- Add tree, form, and search views and menu item for flight.phase.duration model
- Implement constraint to ensure correct event time sequence per phase/flight
- Update README

16.0.1.1.0 version

ayushin commented:
can we think of /consider a solution to calculate durations programmatically without storing them in the database model?

ayushin commented:
further, we should not be hard-coding "A" for only calculating actual durations

annzzu committed:
[IMP] flight_event: phase duration calculation considering time_kind and phases

- add time kinde to phase duration to have each time kind for calculated phase durations
- add func to calculate phases and time kind fields

16.0.1.1.0 version

annzzu commented:
I have already added the `time_kind` field in `flight.phase.duration`, which allows us to retrieve durations filtered by both phase and time kind.

```python
def get_duration(self, phase_name, time_kind):
    """
    Get the duration for a specific phase and time kind.
    Returns the duration or 0 if not found.
    """
    pd_id = self.phase_duration_ids.filtered(lambda pd_id:
                                             pd_id.time_kind == time_kind
                                             and pd_id.phase_id.name == phase_name)

    return pd_id.duration or 0
```

Now, there are no more hardcoded phases or time kinds. You can simply specify the phase and time kind you want, and the corresponding duration will be returned. even better solution than block_a

annzzu commented on flight_event/models/flight_flight.py:
```diff
@@ -29,37 +41,94 @@ def _check_event_sequence(self):
+            if duration_info not in updated_phase_durations:
+                duration.unlink()
+
+    def get_duration(self, phase_name, time_kind):
```
function which calculates duration with phase and time kind

ayushin commented:
please rebase on the current branch so i can review at once

annzzu commented:
This is rebased. you can check

ayushin commented:
closed in favour of complete solution
